cirq-core/cirq/protocols/resolve_parameters.py

```diff
@@ -177,6 +177,10 @@ def resolve_parameters(
     if isinstance(val, (list, tuple)):
         return cast(T, type(val)(resolve_parameters(e, param_resolver, recursive) for e in val))
 
+    is_parameterized = getattr(val, '_is_parameterized_', None)
+    if is_parameterized is not None and is_parameterized() is False:
```

Comment:
nit: Style guide says to use `not` instead of comparing to `False` (https://google.github.io/styleguide/pyguide.html#2144-decision):
```suggestion
    if is_parameterized is not None and not is_parameterized():
```
If if returns `NotImplemented` that is still considered a truthy value, so the condition will not hold and we will proceed to regular parameter resolution as before.